tests/integration/aof.tcl

```diff
@@ -478,22 +478,18 @@ tags {"aof external:skip"} {
     }
 
     test {EVAL timeout with slow verbatim Lua script from AOF} {
-        create_aof $aof_dirpath $aof_file {
-            append_to_aof [formatCommand select 9]
-            append_to_aof [formatCommand eval {redis.call('set',KEYS[1],'y'); for i=1,1500000 do redis.call('ping') end return 'ok'} 1 x]
-        }
-
-        start_server [list overrides [list dir $server_path appendonly no lua-time-limit 1 aof-use-rdb-preamble no]] {
+        start_server [list overrides [list dir $server_path appendonly yes lua-time-limit 1 aof-use-rdb-preamble no]] {
             # generate a long running script that is propagated to the AOF as script
             # make sure that the script times out during loading
+            create_aof $aof_dirpath $aof_file {
+                append_to_aof [formatCommand select 9]
+                append_to_aof [formatCommand eval {redis.call('set',KEYS[1],'y'); for i=1,1500000 do redis.call('ping') end return 'ok'} 1 x]
+            }
             set rd [redis_deferring_client]
-            r config set appendonly yes
             set start [clock clicks -milliseconds]
             $rd debug loadaof
             $rd flush
-            after 100
-            catch {r ping} err
-            assert_match {LOADING*} $err
```

Comment:
please add the PING assertion after the wait_for_log_messages,
the point of the test is to make sure redis is "responsive" to pings during long scripts in aof loading.

Reply:
Test changed, So this is not needed (because PING always ack PONG)

Reply:
i don't understand what you mean.
we're waiting for the server to run a slow script during loading.
then we wanna be sure that it processes commands during a busy script, and it rejects them with LOADING rather than BUSY.

so we're now waiting for the log message, and after that we can add the assertion that we used to have.

Reply:
i still don't see an assertion for `LOADING`.
IIUC, right after (or before) the wait for `Slow script detected`, PING should return LOADING error..

another thing i don't understand.
looking at the code before this change, the test used to enable `apendonly`, which caused a rewrite, so there's no script in the AOF file that's loaded.. so how come the assert for LOADING response succeeded?

p.s. i added this to the list in the top comment, let me know if i'm missing anything.

Reply:
@oranagra If you rollback to the original commit and add an `after 100` just right under the `config set appendonly yes` line you will always get an PONG instead of LOADING.

Hypothesis, not confirmed:
In the original test, the `debug loadaof` is immediately executed just after aofrewrite triggered and before AOF actually rewritten, which means the long script AOF gets loaded before rewritten.

Reply:
@chenyang8094 do you agree that this check is still missing, or is there something i'm missing?

Reply:
@oranagra I think i should use `count_log_message` not `wait_for_log_messages` to check the logs.
Now we are loading AOF in redis startup phase instead of executing `debug loadaof` in running, so once we enter the assertion part of the test, redis must finish AOF load and print "Ready to accept connections" , so, I don't Know why we still need to assert on `PING` at this point, since it definitely returns `PONG`.
```
create_aof $aof_dirpath $aof_file {
            append_to_aof [formatCommand select 9]
            append_to_aof [formatCommand eval {redis.call('set',KEYS[1],'y'); for i=1,1500000 do redis.call('ping') end return 'ok'} 1 x]
        }
        start_server [list overrides [list dir $server_path appendonly yes lua-time-limit 1 aof-use-rdb-preamble no]] {  
            assert_equal [count_log_message 0 "Slow script detected"] 1
            assert_equal [r get x] y
        }
```

Reply:
p.s I see that there are still many tests, using `wait_done_loading` immediately after `start_server` to wait for redis to end loading, I think this is not necessary, am I missing something?

Reply:
i don't think the test infra is waiting for after the loading is complete.
if it did, then we wouldn't be able to test how redis behaves during loading (which is what i'm asking here, to check that PING returns LOADING).

i don't see anything in the test suite that's looking for `Ready to accept connections`.
i see `wait_server_started` which waits for the `PID:` log message (printed before the loading)
and i see `server_is_up` which attempts to connect to the socket (which is obviously already bound, otherwise the -LOADING feature would be dead code, at least on masters).

am i missing anything?

Reply:
The following figure shows the startup log of redis in this test:
![image](https://user-images.githubusercontent.com/13696140/150446387-e5b05390-bb48-43e8-aa65-ea0fea50e0d1.png)

Let's look at the implementation of the start_server function:  

```
proc start_server {options {code undefined}} {
    set previous_ready_count [count_message_lines $stdout "Ready to accept"]
    spawn_server
    while 1 {
          # check that the server actually started and is ready for connections
          if {[count_message_lines $stdout "Ready to accept"] > $previous_ready_count} {
              break
          }
          after 10
    }
     # append the server to the stack
     lappend ::servers $srv

     # connect client (after server dict is put on the stack)
     reconnect

     # execute provided block
      set num_tests $::num_tests
      if {[catch { uplevel 1 $code } error]} {
          # Ignore details
     }
}
```

IIUC  `start_server` will wait for redis to be ready (that is, the` "Ready to accept" `log appears) before executing the code in test, and AOF has already been loaded at this time, isn't it?

Reply:
you're right. i grepped for "Ready to accept connections" so didn't find that one.
thing is that as far as i recall, the problem for which test test was written for (#7870) is when script that's part of an AOF times out and commands are executed. maybe that specific crash would have occurred even if commands are not executed (timeout is enough), but i rather test the other parts of that flow so that we can find other problems or regressions in the future.

so i see two ways to handle that.
1. switch back to using DEBUG LOADAOF (sorry that i asked to move away from it).
2. add some optional argument to start_server that will allow it to run the tests before the server is done loading (i imagine this can come in handy in other tests in the future)

Reply:
i prefer 1,  @oranagra WDYT?

Reply:
yes. let's do that.. (avoid infrastructure changes at this point).
sorry for the back and forth.